### ISSUE TITLE
Improve Optics.Fold docs

### ISSUE THREAD

**Issue by adamgundry** [well-typed/optics#426]
At the moment `Optics.Fold` uses `foldVL` as the canonical introduction form, and has lots of canonical elimination forms. I think it would be clearer to use `folding @[]` instead for the introduction form, and `toListOf` for the elimination form. This makes explicit the connection between `Fold s a` and `s -> [a]`.

**Comment by arybczak** [comment]
IMO `foldVL_` is much easier to grok than `folding` :thinking: Its resemblence to `traversalVL` is analogous to how `traverse_` and `traverse` differ.

**Comment by adamgundry** [comment]
You're right that we should emphasise the relationship between `traversalVL` and `foldVL`. That's helpful if one understands traversals already. But I think that as a route to understanding folds without assuming an understanding of traversals, `s -> [a]` is simpler.

I'm currently working on a workshop about `optics` for ZuriHac, which is why I'm thinking about this. Once that is under control I'll try to make a PR with the doc changes I envision.

**Comment by phadej** [comment]
`foldring` with a stricter signature, i.e.

```haskell
(forall b. (a -> b -> b) -> b -> s -> b) -> Fold s a
```

is IMHO simplest introduction form. Often quite optimal as well.

As I hint in #427, our type for `foldVL` is not entirely correct - and we don't discuss that in the docs:

```haskell
*Optics.Core> :t foldVL traverse
foldVL traverse :: Traversable t => Fold (t a) a
```

works, but i'm not sure it should or shouldn't. (`folding foldMap` of `foldring foldr` would be better, wouldn't it?)

**Comment by arybczak** [comment]
> folding foldMap of foldring foldr would be better, wouldn't it?

These provide different performance characteristics. Last time I checked `foldVL traverse_` was the fastest for most standard containers.
